dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -290,10 +290,13 @@ await foreach (Completions completions in response)
 
             string fqn = functionCallResponse.FullyQualifiedName;
 
-            chatOptions.Messages.Add(resultChoice.Message);
-            chatOptions.Messages.Add(new ChatMessage(ChatRole.Function, serializedFunctionResult) { Name = fqn });
+            if (resultChoice.Message.Content is { Length: > 0 })
```

Comment:
Why is this being agreed only conditionally? I thought it was always required and without it subsequent calls will fail... is that not our no longer the case?

Reply:
At first I thought it was the problem (adding an empty assistant message to the history) in the Beta 10 bug, and I just added the message when the assistant actually render a message. 

I kept this change in place to see if it was actually impacting the results and running the Example 59, it worked as expected.